File: app/src/test/resources/testcode-php/ClassUsagePatterns.php
<?php

// Test file to verify PHP class usage patterns

class BaseClass
{
    public static function staticMethod(): string
    {
        return "static";
    }

    public function instanceMethod(): string
    {
        return "instance";
    }
}

class ExtendedClass extends BaseClass
{
    private BaseClass $field;

    public function __construct(BaseClass $param)
    {
        // Constructor call
        $this->field = new BaseClass();

        // Static access
        BaseClass::staticMethod();
    }

    // Method with type hints
    public function processBase(BaseClass $input): BaseClass
    {
        return $input;
    }

    // Return type hint
    public function getInstance(): BaseClass
    {
        return $this->field;
    }
}

class UsageExample
{
    // Variable with type hint
    private BaseClass $myBase;

    public function __construct()
    {
        $this->myBase = new BaseClass();
    }
}

File: app/src/test/resources/testcode-py/class_usage_patterns.py
"""Test file to verify Python class usage patterns"""

class BaseClass:
    @staticmethod
    def static_method():
        return "static"

    def instance_method(self):
        return "instance"


class ExtendedClass(BaseClass):
    """Class extending BaseClass"""

    def __init__(self, param: BaseClass):
        super().__init__()
        # Constructor call
        self.field: BaseClass = BaseClass()
        # Static access
        result = BaseClass.static_method()

    # Type hints
    def process_base(self, input: BaseClass) -> BaseClass:
        return input

    # Return type annotation
    def get_instance(self) -> BaseClass:
        return self.field


# Variable with type hint
my_base: BaseClass = BaseClass()

# Import pattern
from typing import List
instances: List[BaseClass] = []
